### DISCUSSION TITLE
Is it possible to have two docs plugins with nested base paths?

### DISCUSSION THREAD

**Discussion by jakub-cribl** [facebook/docusaurus#6104] [Q&A] [answered]
I created a page with two docs plugins: `docs` and `release-notes`, the first one will be versioned, the second one won't.

When the `release-notes` has `routeBasePath: 'release-notes'` everything works, but when I change it to `docs/release-notes` every release note URL shows a "Page not found" error.

When I had the release notes as a blog plugin, the path nesting worked.

Is there a way to make it work for docs, is it a bug or a limitation of Docusaurus?

PS. I created a repo with the config - a fresh Docusaurus page with the docs copied to simulate the release notes - https://github.com/jakub-cribl/docusaurus-multi-docs

**Comment by slorber** [comment, marked as answer]
The problem is because the default docs instance is using `/docs/` and the 2nd one is using `/docs/release-notes`

We use catch-all routes like `/docs/*`, and this probably shadows the `/docs/release-notes/*` route. Try using something else like `/myDocs/release-notes` and it should work.

We have some things that I don't like much currently in our routing infra, and we should definitively allow this to work despite one base path being included into another. 

We'll upgrade to React-Router 6 someday (https://github.com/facebook/docusaurus/pull/6037) and it has a new route ranking logic, and it may solve this specific problem. But afaik it won't solve all our problems. I believe it should be possible to use Docusaurus with all plugins using routeBasePath = `/` and it should work fine. Current implementation and even RR6 probably won't help, we need something better.

**Reply by jakub-cribl** [reply]
The funny thing is that currently I have a docs plugin with routeBasePath = `docs` and a blog plugin with routeBasePath = `docs/release-notes` and it works.

Wanted to move the release notes into docs because it would make more sense with how it is configured now, but it seems that I'll need to wait and see after RR is updated.

Thanks for the answer!

**Reply by slorber** [reply]
So, I believe that it's actually the blog path the shadows the docs path, as both are the same.

As far as I remember we implemented something so that paths with more segment come first in the matching algorithm. The problem is probably using 2 plugins with the exact same base path 😓 changing the blog path should make the docs work

**Reply by Josh-Cena** [reply]
@slorber I hope that upgrading to RR v6 would mitigate that problem. Seems they have a smarter route matching algorithm.

**Reply by slorber** [reply]
I don't think so, last time I took a look at this, the algo was like the Reach one and just sort routes before doing the matching. What we need is enter a parent route only if a subroute match.
